apps/ewallet/lib/ewallet/web/paginators/start_after_paginator.ex

```diff
@@ -0,0 +1,340 @@
+    start_by = map_atom_attr(attrs, "start_by", default_field, default_mapped_fields)
+
+    case is_allowed_start_by(start_by, allowed_fields) do
+      true ->
+        attrs =
+          attrs
+          |> Map.put("sort_by", sort_by)
+          |> Map.put("start_by", start_by)
+
+        paginate(queryable, attrs, repo)
+
+      _ ->
+        available_fields =
+          allowed_fields
+          |> Enum.map(&Atom.to_string/1)
+          |> Enum.join(", ")
+
+        start_by = Atom.to_string(start_by)
+
+        msg =
+          "start_by: `#{start_by}` is not allowed. The available fields are: [#{available_fields}]"
+
+        {:error, :invalid_parameter, msg}
+    end
+  end
+
+  def map_atom_attr(attrs, key, default, mapping) do
+    attrs[key]
+    |> map_default(default)
+    |> map_field(mapping)
+    |> map_atom(default)
+  end
+
+  defp map_default(original, default) do
+    case original do
+      nil -> default
+      any -> any
+    end
+  end
+
+  defp map_field(original, mapping) do
+    case mapping[original] do
+      nil -> original
+      mapped -> mapped
+    end
+  end
+
+  defp map_atom(original, default) do
+    String.to_existing_atom(original)
+  rescue
+    ArgumentError -> default
+  end
+
+  @doc """
+  Paginate a query using the given `start_after` and `start_by`.
+  Returns a paginator with all records after the specify `start_after` record.
+  """
+  def paginate(queryable, attrs, repo \\ Repo)
+
+  # Query and returns `Paginator`
+  def paginate(
+        queryable,
+        %{
+          "start_by" => start_by,
+          "start_after" => {:ok, start_after},
+          "per_page" => per_page
+        } = attrs,
+        repo
+      ) do
+    attrs = Map.put(attrs, "start_after", start_after)
+
+    {records, more_page} =
+      queryable
+      |> get_queryable_start_after(attrs)
+      |> get_queryable_order_by(attrs)
+      |> fetch(per_page, repo)
+
+    pagination = %{
+      per_page: per_page,
+      start_by: Atom.to_string(start_by),
+      start_after: start_after,
+      # It's the last page if there are no more records
+      is_last_page: !more_page,
+      count: length(records)
+    }
+
+    %Paginator{data: records, pagination: pagination}
+  end
+
+  def paginate(queryable, %{"start_after" => nil} = attrs, repo),
+    do: paginate(queryable, %{attrs | "start_after" => {:ok, nil}}, repo)
+
+  # Checking if the `start_after` value exist in the db.
+  def paginate(
+        queryable,
+        %{
+          "start_by" => start_by,
+          "start_after" => start_after,
+          "per_page" => _
+        } = attrs,
+        repo
+      ) do
+    # Verify if the given `start_after` exist to prevent unexpected result.
+    condition =
+      build_start_after_condition(%{"start_by" => start_by, "start_after" => start_after})
+
+    pure_queryable = exclude(queryable, :where)
+
+    try do
+      case repo.get_by(pure_queryable, condition) do
```

Comment:
Sorry I just realized that this puts the whole `case` clause under try/rescue and there's a lot that can happen with `_ -> paginate(...)`.

Can you try/rescue only the `repo.get_by...`?